src/widgets/NavigationBar.js

```diff
@@ -10,35 +10,95 @@ import { Platform, TouchableOpacity, View, StyleSheet } from 'react-native';
 import Icon from 'react-native-vector-icons/FontAwesome';
 
 import { textStyles } from '../globalStyles';
+import { BadgeSet } from './BadgeSet';
+import { generalStrings } from '../localization';
 
-export const NavigationBar = ({ onPressBack, LeftComponent, CentreComponent, RightComponent }) => (
-  <View style={localStyles.container}>
-    <View style={localStyles.leftSection}>
-      <TouchableOpacity onPress={onPressBack} style={localStyles.backButton}>
-        {onPressBack && <Icon name="chevron-left" style={localStyles.backIcon} />}
-      </TouchableOpacity>
-      {LeftComponent && <LeftComponent />}
-    </View>
-    <View style={localStyles.centreSection}>{CentreComponent && <CentreComponent />}</View>
-    <View style={localStyles.rightSection}>{RightComponent && <RightComponent />}</View>
-  </View>
-);
+export class NavigationBar extends React.Component {
+  state = {
+    badge: [{ title: '', type: 'unfinalised', Count: 0 }],
+  };
+
+  routeList = {
+    customerRequisitions: 'ResponseRequisition',
+    supplierRequisitions: 'RequestRequisition',
+    supplierInvoices: 'SupplierInvoice',
+    stocktakes: 'Stocktake',
+    customerInvoices: 'CustomerInvoice',
+  };
+
+  componentWillReceiveProps(props) {
+    if (props.routeName in this.routeList) {
+      const dataType = this.getDataTypeFromRouteName(props);
+      this.refreshData(dataType);
+    }
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  getDataTypeFromRouteName(props) {
+    return props.routeName in this.routeList ? this.routeList[props.routeName] : '';
+  }
```

Comment:
cool helper! Could go in `./src/navigation/selectors` and you could just pass `routeNameDataType` mapped from state to props (in `MenuPage` or connect `NavigationBar`)

Reply:
Done with 14aa68b

Reply:
Refactored with 7b3ab4b